tests/integration/test_charm.py

```diff
@@ -94,395 +68,280 @@ async def test_deploy_s3_integrator(ops_test: OpsTest, charm_versions, s3_bucket
-    namespace = ops_test.model.name
-    sa_name = "custom-sa"
-
-    # Adding a custom property via Spark8t to the service account
-    assert (
-        subprocess.run(
-            [
-                "python",
-                "-m",
-                "spark8t.cli.service_account_registry",
-                "create",
-                "--username",
-                sa_name,
-                "--namespace",
-                namespace,
-                "--conf",
-                "spark.kubernetes.executor.request.cores=0.1",
-                "--conf",
-                "spark.executor.instances=3",
-            ]
-        ).returncode
-        == 0
-    )
-
-    logger.info("Changing configuration for kyuubi-k8s charm...")
-    await ops_test.model.applications[APP_NAME].set_config({"service-account": sa_name})
-
-    logger.info("Waiting for kyuubi-k8s app to be idle...")
-    await ops_test.model.wait_for_idle(
-        apps=[APP_NAME],
-        status="active",
-        timeout=1000,
-    )
-    kyuubi_leader = await find_leader_unit(ops_test, app_name=APP_NAME)
-    assert kyuubi_leader is not None
-
-    logger.info("Running action 'get-password' on kyuubi-k8s unit...")
-    action = await kyuubi_leader.run_action(
-        action_name="get-password",
-    )
-    result = await action.wait()
-
-    password = result.results.get("password")
-    logger.info(f"Fetched password: {password}")
-
-    username = "admin"
-    assert await validate_sql_queries_with_kyuubi(
-        ops_test=ops_test, username=username, password=password
-    )
-
-    # Check exactly 3 executor pods were created.
-    list_pods_process = subprocess.run(
-        ["kubectl", "get", "pods", "-n", namespace, "--sort-by", ".metadata.creationTimestamp"],
-        capture_output=True,
-    )
-
-    assert list_pods_process.returncode == 0
-
-    pods_list = list_pods_process.stdout.decode().splitlines()
-
-    driver_pod_name = ""
-    executor_pod_names = []
-
-    # Last 4 pods in the list are of interest,
-    # one is the driver and 3 should be executor pods
-    for pod in pods_list[-4:]:
-        name = pod.split()[0]
-        if "driver" in name:
-            driver_pod_name = name
-        else:
-            executor_pod_names.append(name)
-
-    expected_executor_pod_names = [
-        driver_pod_name.replace("driver", "exec-1"),
-        driver_pod_name.replace("driver", "exec-2"),
-        driver_pod_name.replace("driver", "exec-3"),
-    ]
-
-    assert len(executor_pod_names) == len(expected_executor_pod_names)
-    assert set(executor_pod_names) == set(expected_executor_pod_names)
+    password = fetch_password(juju)
+    assert validate_sql_queries_with_kyuubi(juju, username=username, password=password)
+
+
+# TODO: rewrite
+# @pytest.mark.skip(reason="This tests need re-write and fixes on integration hub level")
+# def test_read_spark_properties_from_secrets(ops_test: OpsTest):
```

Comment:
In my opinion, we can remove this test altogether because the thing it is trying to test is already obsolete in our case. The Kyuubi charm no longer gets notified of the changes performed at user level using `spark8t` tool, but is concerned with only cluster level configurations provided to it by the Integration Hub. As far as the test that Kyuubi is notified of the cluster level configurations, we already do that in this file. As a result, I think we can drop this function altogether.

Reply:
Agreed, dropped in 0c746fa2139d38481f7917752d5f2b653c5dcc2c